File: servico-musica-java/java.soap/src/main/java/servico/java/soap/config/BancoInit.java
package servico.java.soap.config;

import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import servico.java.soap.entity.Musica;
import servico.java.soap.entity.Playlist;
import servico.java.soap.entity.Usuario;
import servico.java.soap.repository.MusicaRepository;
import servico.java.soap.repository.PlaylistRepository;
import servico.java.soap.repository.UsuarioRepository;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

@Configuration
public class BancoInit {

    @Bean
    CommandLineRunner commandLineRunner(MusicaRepository musicaRepository, PlaylistRepository playlistRepository, UsuarioRepository usuarioRepository) {

        return args -> {
            // popular musica repo
            for (int i = 0; i < 200; i++) {
                musicaRepository.save(
                        new Musica(
                                String.format("Musica %d",i),
                                String.format("Artista %d",i)
                        )
                );
            }

            // popular usuario repo
            int min = 16;
            int max = 45;
            short idade;

            for (int i = 0; i < 200; i++) {
                idade = (short) (min + (int)(Math.random() * ((max - min) + 1)));
                usuarioRepository.save(
                        new Usuario(
                                String.format("Usuario %d",i),
                                idade
                        )
                );
            }

            // popular playlist usuario
            List<Musica> musicas = musicaRepository.findAll();

            for (Usuario usuario : usuarioRepository.findAll()) {

                Playlist playlist = new Playlist(
                        String.format("playlist do usuario de id %d", usuario.getId())
                );

                playlist.setUsuario(usuario);

//                for (Musica musica : musicas) {
//                    playlist.getMusicas().add(musica);
//                }

                List<Musica> copia = new ArrayList<>(musicas);
                Collections.shuffle(copia);

                playlist.getMusicas().addAll(copia.subList(0, 3));

                playlistRepository.save(playlist);
            }
        };
    }
}

File: servico-musica-kotlin/kotlin.soap/src/main/kotlin/servico/kotlin/soap/config/BancoInit.kt
package servico.kotlin.soap.config

import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import servico.kotlin.soap.entity.Musica
import servico.kotlin.soap.entity.Playlist
import servico.kotlin.soap.entity.Usuario
import servico.kotlin.soap.repository.MusicaRepository
import servico.kotlin.soap.repository.PlaylistRepository
import servico.kotlin.soap.repository.UsuarioRepository

@Configuration
class BancoInit {

    @Bean
    fun commandLineRunner(
        musicaRepository: MusicaRepository,
        playlistRepository: PlaylistRepository,
        usuarioRepository: UsuarioRepository
    ): CommandLineRunner {

        val tamanhoBd = 199
        return CommandLineRunner {

            // músicas
            for (i in 0..tamanhoBd) {
                musicaRepository.save(
                    Musica(
                        "Musica $i",
                        "Artista $i"
                    )
                )
            }

            // usuários
            val min = 16
            val max = 45

            for (i in 0..tamanhoBd) {
                val idade = (min..max).random().toShort()

                usuarioRepository.save(
                    Usuario(
                        "Usuario $i",
                        idade
                    )
                )
            }

            // playlists
            val musicas = musicaRepository.findAll()

            for (usuario in usuarioRepository.findAll()) {

                val playlist = Playlist(
                    "playlist do usuario de id ${usuario.id}"
                )

                playlist.usuario = usuario

                val copia = musicas.toMutableList()
                copia.shuffle()

                playlist.musicas.addAll(copia.take(3))

                playlistRepository.save(playlist)
            }
        }
    }
}